### PR TITLE
Remove /admin route

### PR DESCRIPTION
**Pull request by jeffsee55**
This preview pattern is slightly different than what we've attempted in the past in that with additional constraints like no inline-editing support we're able to fully remove Tina from being a page-level concern and if we like it we wouldn't need `admin` routes at all. This allows us to stay out of the way of the developer more with a couple of tradeoffs:

- This pattern discourages "downstream" Tina configurations. Meaning if you're on the `posts/[filename].tsx` and you want to add a `Content Creator`, you're left fumbling around with dynamic imports on your own. (Note that any downstream reference to the Tina CMS context will blow up on you when `preview` is false).
- @logan-anderson mentioned there's some extra overhead that comes with using `preview` mode. Essentially the fact that you'll need to bring a server with you instead of being able to drop this fully-static site in an s3 bucket. And what's important to note is that I stumbled into this with preview mode but Logan is going to toy around with using a more generic cookie or localStorage solution instead which would work with any framework

### CODE REVIEW

File: pages/posts/[filename].tsx [hunk 1]
```diff
@@ -35,17 +37,19 @@ export const query = (gql) => gql`
...
-      variables: { relativePath: `${params.filename}.md` },
-    }),
+    props: {
+      data: await client.request<{ getPostsDocument: Posts_Document }>(query, {
+        variables,
+      }),
+      variables,
+      query,
+      preview: !!preview,
+    },
```
Comment (jeffsee55) [inline pages/posts/[filename].tsx:52]: This shape is required for _all_ `getStaticProps` return values. Perhaps we could be a helper function or typescript type

File: pages/api/preview.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export default function handler(req, res) {
+  const isPreview = req.preview;
+
+  if (isPreview) {
+    res.clearPreviewData();
+  } else {
+    res.setPreviewData({});
+  }
+  res.redirect("/");
```
Comment (jeffsee55) [inline pages/api/preview.js:9]: @logan-anderson says there's a clean way to grab the referrer URL and send the user back to it 👍

**Comment by logan-anderson** [comment]
@jeffsee55  I was able to remove nextjs preview mode from this implementation see https://github.com/tinacms/tina-cloud-starter/pull/60

**Comment by jeffsee55** [comment]
Closing in favor of https://github.com/tinacms/tina-cloud-starter/pull/60. Not sure we'll merge any of this just yet regardless, the "admin" pattern seems to put people's minds at ease that they won't be loading Tina in production views so even though this still works we should get some feedback from others

**Comment by jeffsee55** [comment]
Merged in Logan's changes, much better since we weren't really using preview mode for it's actual purpose.

**Comment by logan-anderson** [comment]
Issue here https://github.com/tinacms/tina-cloud-starter/issues/63

@jeffsee55 @DirtyF  @dwalkr @Enigmatical I updated the Readme and added some comments in the code wouldn’t mind a second set of eyes.

File: pages/_app.js [hunk 3]
```diff
@@ -1,30 +1,44 @@
...
-   * If the route starts with /admin, we'll wrap the entire component tree
-   * with Tina, meaning your non-admin routes won't contain any Tina code
-   * in their bundles.
-   */
-  if (route.startsWith("/admin")) {
+// InnerApp that handles rendering edit mode or not
+function InnerApp({ Component, pageProps }) {
+  const {edit} = useEditState()
+  if (edit) {
+    // Dynamically load tina ONLY when in edit mode so it does not effect the bundle
```
Comment (DirtyF) [inline pages/_app.js:10]:
```suggestion
    // Dynamically load Tina only when in edit mode so it does not affect production
```

File: pages/_app.js [hunk 4]
```diff
@@ -42,6 +56,14 @@ function MyApp({ Component, pageProps }) {
       `}</style>
     </>
   );
+};
+
+
+// Return the app wrapped in the edit provider
```
Comment (DirtyF) [inline pages/_app.js:62]:
```suggestion
// Our app is wrapped with edit provider
```

**Comment by Enigmatical** [comment]
Got a handful of conflicts on here (nothing serious, I suspect).  Once those are resolved, I'm happy to take a look at the finished PR 👍

**Comment by logan-anderson** [comment]
> Got a handful of conflicts on here (nothing serious, I suspect). Once those are resolved, I'm happy to take a look at the finished PR 👍

@Enigmatical  Should be good now

File: components/Spinner/index.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+    </div>
+  );
+};
+
+export const LoadingPage: React.FC<{ text?: string }> = ({
+  text = "Wait a bit, Tina is loading data from GitHub",
```
Comment (Enigmatical) [inline components/Spinner/index.tsx:15]: I'm assuming this shows when you are working locally, as well?  Do we just want to truncate the "from GitHub" to make it agnostic?

"Wait a bit, Tina is loading data"

File: utils/editState.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
+// This file handles whether or not a user is in edit mode
+
+import React, { useContext, useState } from 'react';
+const LOCALSTORAGEKEY = 'tinaisadmin'
```
Comment (Enigmatical) [inline utils/editState.tsx:4]: Not a blocker, but we might want to better namespace this?  `tinacms.isEditing` or `tina.isEditing`?  I doubt we're going to brush up against existing keys and I don't know how much LOCALSTORAGE is used elsewhere.

File: utils/editState.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        return isEdit && isEdit === 'true'
+    }
+    // assume not editing if SSR
+    return false
+}
+
+export const setEditing = (isEditing: boolean) => {
+    if(!isSSR){
+        console.log('setting edit local storage')
+        console.log(isEditing?  'true': 'false')
```
Comment (Enigmatical) [inline utils/editState.tsx:21]: Probably safe to remove these.

**Review by Enigmatical — CHANGES_REQUESTED** [review]
I left some comments that can be pretty quickly addressed.  I have a bigger one, but I'll bring that up in the #oss channel.

**Comment by logan-anderson** [comment]
@Enigmatical Thanks for the feedback I have updated based on your comments.

**Review by Enigmatical — APPROVED** [review]

**Pull request merged**